### PR TITLE
reset to pre-checkout state by memory

### PR DESCRIPTION
**Pull request by jlaswell**
Brought this back up to speed as best as I could remember. I also removed some of the unused comments that I saw were duplicated or ideas that we've moved away from using.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
     "autoload": {
         "psr-4": {
-            "Ericabwalker\\PHPfinal\\": "src",
-            "Ericabwalker\\PHPfinal\\Controllers\\": "resources"
+            "Ericabwalker\\PHPfinal\\": "src"
```
Comment (jlaswell) [inline composer.json:4]: Switching back to a single namespace in the `src` directory.

```
src
├── Controllers
│   ├── bookscontroller.php # There is a note about this file addition below.
│   └── controller.php
├── Models
│   └── book.php
└── router.php

2 directories, 4 files
```

File: docker-compose.yaml [hunk 2]
```diff
@@ -1,33 +1,20 @@
 version: "3.7"
 
 services:
-    hello-container:
```
Comment (jlaswell) [inline docker-compose.yaml:4]: Re-removing these services since we mentioned they aren't needed.

File: docker-compose.yaml [hunk 3]
```diff
@@ -36,18 +23,8 @@ services:
             MYSQL_PASSWORD: "secret"
             MYSQL_ROOT_PASSWORD: "secret"
         ports:
-            - "3306:3306"
+            - "3307:3306"
```
Comment (jlaswell) [inline docker-compose.yaml:26]: This was conflicting with the local database, so we adjusted it to a non-default MySQL port.

File: phpcs.xml.dist [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
```
Comment (jlaswell) [inline phpcs.xml.dist:1]: This is just a common linting file that we use in all of our projects. It doesn't affect the behavior at all and is just a quality of life addition.

File: src/Controllers/bookscontroller.php [hunk 5]
```diff
@@ -59,6 +55,6 @@ function update_book($title, $author, $pages, $category, $bookID=null)
     }
 
     function add_book_form() {
-        
+        echo "Add form will go here";
```
Comment (jlaswell) [inline src/Controllers/bookscontroller.php:58]: We added some output here to try and test getting this method to be called.

File: src/Controllers/bookscontroller.php [hunk 6]
```diff
@@ -1,11 +1,7 @@
 <?php
```
Comment (jlaswell) [inline src/Controllers/bookscontroller.php:1]: Notice that I've renamed the `controller.php` file to `bookscontroller.php`. The filename being different from the classname is the reason we could not autoload the `BooksController` into the `Router`! When preparing autoload files, composer will look for classes based on filename. Since the file was named `controller.php`, but the class is `BooksController`, they don't match and the file cannot be loaded into other classes.

tl;dr A class's name must match the filename when autoloading using psr-4.

File: src/Controllers/controller.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
+<?php
```
Comment (jlaswell) [inline src/Controllers/controller.php:1]: I left this file in here as reference. We can remove this without issue.

File: src/router.php [hunk 8]
```diff
@@ -1,32 +1,23 @@
...
-            if($method == $_SERVER['REQUEST_METHOD']) {
-                var_dump($route);
+    public function execute($routes)
+    {
+        $method = $_SERVER['REQUEST_METHOD'];
+        foreach ($routes[$method] as $uri => $action) {
+            if ($uri == $_SERVER['REQUEST_URI']) {
+                $method = explode('@', $action)[1];
+                $controller = new BooksController();
+                $controller->$method();
```
Comment (jlaswell) [inline src/router.php:19]: You remembered that `explode` method is like a string split function in PHP, so we are using the first index, or right section, of `BooksController@add_books_form` as the `$method`. Then we call that method name using the `$controller->$method();` notation just like we saw in the `Request` class from other projects.
